app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/StockHelper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Plugin\CatalogInventory;
+
+use Magento\Catalog\Model\ResourceModel\Product\Link\Product\Collection;
+use Magento\CatalogInventory\Helper\Stock;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\InventoryCatalog\Model\GetStockIdForCurrentWebsite;
+use Magento\InventoryIndexer\Indexer\IndexStructure;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
+
+/**
+ * Class provides around Plugin on Magento\CatalogInventory\Helper\Stock::addInStockFilterToCollection
+ */
+class StockHelper
+{
+
+    /**
+     * @var GetStockIdForCurrentWebsite
+     */
+    private $getStockIdForCurrentWebsite;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @param GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite
+     * @param ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(
+        GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite,
+        ScopeConfigInterface $scopeConfig,
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+    ) {
+        $this->getStockIdForCurrentWebsite = $getStockIdForCurrentWebsite;
+        $this->scopeConfig = $scopeConfig;
+        $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+    }
+
+    /**
+     * Plugin to use multi stocks for filtering the collection to return only in stock products.
+     *
+     * @param Stock $subject
+     * @param callable $proceed
+     * @param Collection $collection
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function aroundAddInStockFilterToCollection(Stock $subject, callable $proceed, $collection)
```

Comment:
We need to take into account all of possible configuration states related to `can manage stock`
1) Product has `use_config_manage_stock = 1` (use global configuration). And global option `manage_stock=1`
2) Product has `use_config_manage_stock = 1` (use global configuration). And global option `manage_stock=0`
3) Product has `use_config_manage_stock = 0` (use global configuration). And local (not global) option `manage_stock=0`
4) Product has `use_config_manage_stock = 0` (use global configuration). And local (not global) option `manage_stock=1`

All of this cases should be covered with integration tests